### ISSUE TITLE
Missing state/context in callback call. 

### ISSUE THREAD

**Issue by jellonek** [gragas/simpletcp#4]
For this moment callback function is called with client ip, write queue and received data, so callback function, if needs to track connection state, has no such possibility. IP address is not enough (there can be several connections from one IP in different states).

Quick solution to this would be storing client ip in context dictionary and then passing this dict object to callback function, so it could modify/operate on this dict later, having same object accessible between calls to callback.

**Comment by gragas** [comment]
Good point. I'll look into this sometime this week. Feel free to fork and submit a pull request if you come up with a solution you like.
